### ISSUE TITLE
Patterns with trailing newlines are not parsed in golang

### ISSUE THREAD

**Issue by ievans** [returntocorp/semgrep#191]
Patterns with trailing newlines are not parsed in golang

A yaml file multiline pattern by default uses the `|` operator which (unlike `|-`) does not trim trailing newlines [1]. I verified this specifically is causing the golang pattern parser to choke, eg:

'$TOKEN := jwt.NewWithClaims(...)' <- validates
'$TOKEN := jwt.NewWithClaims(...)\n' <- fails parsing

(https://stackoverflow.com/a/21699210)

Another example:
```
in go/deadcode/useless-ifelse.yaml, pattern in rule useless-if-conditional can't be parsed for language go: if ($X) {
    ...
} else if ($X) {
    ...
}

in go/deadcode/useless-ifelse.yaml, pattern in rule useless-if-body can't be parsed for language go: if ($X) {
    $S
} else {
    $S
}
```

**Comment by ievans** [comment]
Comments from @aryx: Newlines have a meaning in go; Go inserts automatically some semicolons when it encounters certain newlines.  But in sgrep mode it does not in certain conditions, because i donr want $X==$X to be parse with an ending semicolon

Proposal: perhaps sgrep-lint should remove the final trailing newline automatically so that the user doesn't have to understand the difference between `|` and `|-` in yaml

**Comment by dlukeomalley** [comment]
TIL I learned the difference between `|` and `|-`. This point seems subtle and likely to cause issues in the future. Do you still feel good about your proposal?

This website has a nice demo: https://yaml-multiline.info/

**Comment by ievans** [comment]
Confirmed this is fixed in 0.9.0 or earlier! https://semgrep.live/ZqvA

Thanks for the prod @dlukeomalley . Closing.
